testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/elytron/SlaveHostControllerElytronAuthenticationTestCase.java

```diff
@@ -118,13 +118,15 @@ private void slaveWithPlainMechanism() throws Exception {
 
         // Reload the slave and check that it produces a registered slave
         reloadSlave();
+        testSupport.getDomainSlaveLifecycleUtil().awaitHostController(System.currentTimeMillis());
         readHostControllerStatus(getDomainMasterClient());
 
         // Set the allowed mechanism back to Digest-MD5
         getDomainMasterClient().execute(changePresentedMechanisms("master", new HashSet<>(Arrays.asList("DIGEST-MD5"))));
         getDomainSlaveClient().execute(changeSaslMechanism("slave", "DIGEST-MD5"));
         reloadSlave();
         testSupport.getDomainSlaveLifecycleUtil().awaitHostController(System.currentTimeMillis());
+        assertTrue("Host \"slave\" did not reconnect to master with plain mechanism.", hostRunning(FAILED_RELOAD_TIMEOUT_MILLIS));
         readHostControllerStatus(domainMasterClient);
```

Comment:
Yeah, agreed on the redundancy, I should have removed that one earlier.

Good point on the other test, I'd been focusing on the broken one, but I'd wondered about that when I'd look at it before. I'll have a think about how we could do something better.